.github/workflows/build.yml

```diff
@@ -1,13 +1,290 @@
-name: CI
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
 
-on: [push]
+name: "Build"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
 
 jobs:
-  build:
+  tests:
+    name: "Test"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies: ["lowest", "highest"]
+        php-version:
+          - "7.2"
+          - "7.3"
+          - "7.4"
+        operating-system: ["ubuntu-latest", "windows-latest"]
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2.0.0"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@1.8.2"
+        with:
+          coverage: "none"
+          php-version: "${{ matrix.php-version }}"
+          extensions: mbstring
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v1.1.2"
+        with:
+          path: "~/.composer/cache"
+          key: "php-${{ matrix.php-version }}-composer-${{ hashFiles('**/composer.json') }}"
+          restore-keys: "php-${{ matrix.php-version }}-composer-"
```

Comment:
`composer.json` changes rarely: we probably don't care about this cache key/restore key at all?

Reply:
Done.